### ISSUE TITLE
X11 problem with 4.11.0-kfd kernel

### ISSUE THREAD

**Issue by oheid** [RadeonOpenCompute/ROCK-Kernel-Driver#29]
The latest 4.11.0-kfd kernel has a problem with X on my two Ryzen 1700X / Radeon 480 boxes (running Fedora 26). There is no such issue with the previous 4.9.0-kfd+ kernel. Xorg.0.log says:
```
...
[    19.157] (II) AMDGPU: Driver for AMD Radeon:
	All GPUs supported by the amdgpu kernel driver
[    19.160] (II) [KMS] drm report modesetting isn't supported.
[    19.160] (EE) Screen 0 deleted because of no matching config section.
[    19.160] (II) UnloadModule: "amdgpu"
[    19.160] (EE) Device(s) detected, but none match those in the config file.
[    19.160] (EE) Fatal server error:
[    19.160] (EE) no screens found(EE) 
...
```
xorg.conf  Device section:
```
...
Section "Device"
	Identifier  "AMD"
	Driver      "amdgpu"
	Option	"AccelMethod" "glamor"
      	Option	"DRI3" "1"
      	Option	"TearFree" "on"
EndSection
...
```
The amdgpu and amdkfd modules are present:
```
# lsmod | grep amd
amdkfd                208896  1
amd_iommu_v2           20480  1 amdkfd
amdgpu               2490368  0
edac_mce_amd           28672  0
ttm                    98304  1 amdgpu
drm_kms_helper        143360  1 amdgpu
kvm_amd              2179072  0
kvm                   581632  1 kvm_amd
drm                   344064  3 amdgpu,ttm,drm_kms_helper
gpio_amdpt             16384  0
gpio_generic           16384  1 gpio_amdpt
i2c_algo_bit           16384  2 igb,amdgpu
```
dmesg says
```
...
[    6.330267] [drm] amdgpu kernel modesetting enabled.
[    6.337185] AMD IOMMUv2 driver by Joerg Roedel <jroedel@suse.de>
[    6.382485] Parsing CRAT table with 0 nodes
[    6.382540] Virtual CRAT table created for CPU
[    6.382592] Parsing CRAT table with 1 nodes
[    6.382645] Creating topology SYSFS entries
[    6.382706] Topology: Add CPU node
[    6.382756] Finished initializing topology
[    6.384510] kfd kfd: Initialized module
[    6.384928] [drm] initializing kernel modesetting (POLARIS10 0x1002:0x67DF 0x148C:0x2372 0xC7).
[    6.385005] [drm] register mmio base: 0xFE900000
[    6.385058] [drm] register mmio size: 262144
[    6.385117] [drm] probing gen 2 caps for device 1022:1453 = 733903/e
[    6.385174] [drm] probing mlw for device 1022:1453 = 733903
[    6.385234] [drm] UVD is enabled in VM mode
[    6.385286] [drm] VCE enabled in VM mode
[    6.385508] amdgpu 0000:29:00.0: Invalid PCI ROM header signature: expecting 0xaa55, got 0xffff
[    6.385582] ATOM BIOS: xxx-xxx-xxx
[    6.385637] [drm] GPU post is not needed
[    6.385699] [drm] vm size is 128 GB, block size is 14-bit
[    6.386535] amdgpu 0000:29:00.0: VRAM: 8192M 0x000000F400000000 - 0x000000F5FFFFFFFF (8192M used)
[    6.386601] amdgpu 0000:29:00.0: GTT: 256M 0x0000000000000000 - 0x000000000FFFFFFF
[    6.386665] [drm] Detected VRAM RAM=8192M, BAR=256M
[    6.386718] [drm] RAM width 256bits GDDR5
[    6.386817] [TTM] Zone  kernel: Available graphics memory: 30818988 kiB
[    6.386876] [TTM] Initializing pool allocator
[    6.386931] [TTM] Initializing DMA pool allocator
[    6.386996] [drm] amdgpu: 8192M of VRAM memory ready
[    6.387051] [drm] amdgpu: 32103M of GTT memory ready.
[    6.387113] [drm] GART: num cpu pages 65536, num gpu pages 65536
[    6.387239] [drm] PCIE GART of 256M enabled (table at 0x000000F400040000).
[    6.387341] amdgpu 0000:29:00.0: amdgpu: using MSI.
[    6.387409] [drm] amdgpu: irq initialized.
[    6.558790] amdgpu: [powerplay] amdgpu: powerplay sw initialized
[    6.562838] amdgpu 0000:29:00.0: fence driver on ring 0 use gpu addr 0x0000000000400008, cpu addr 0xffff8807e4c8a008
[    6.562944] amdgpu 0000:29:00.0: fence driver on ring 1 use gpu addr 0x0000000000400018, cpu addr 0xffff8807e4c8a018
[    6.563043] amdgpu 0000:29:00.0: fence driver on ring 2 use gpu addr 0x0000000000400028, cpu addr 0xffff8807e4c8a028
[    6.563140] amdgpu 0000:29:00.0: fence driver on ring 3 use gpu addr 0x0000000000400038, cpu addr 0xffff8807e4c8a038
[    6.563234] amdgpu 0000:29:00.0: fence driver on ring 4 use gpu addr 0x0000000000400048, cpu addr 0xffff8807e4c8a048
[    6.563332] amdgpu 0000:29:00.0: fence driver on ring 5 use gpu addr 0x0000000000400058, cpu addr 0xffff8807e4c8a058
[    6.563423] amdgpu 0000:29:00.0: fence driver on ring 6 use gpu addr 0x0000000000400068, cpu addr 0xffff8807e4c8a068
[    6.563519] amdgpu 0000:29:00.0: fence driver on ring 7 use gpu addr 0x0000000000400078, cpu addr 0xffff8807e4c8a078
[    6.563610] amdgpu 0000:29:00.0: fence driver on ring 8 use gpu addr 0x0000000000400088, cpu addr 0xffff8807e4c8a088
[    6.563691] amdgpu 0000:29:00.0: fence driver on ring 9 use gpu addr 0x000000000040009c, cpu addr 0xffff8807e4c8a09c
[    6.564062] amdgpu 0000:29:00.0: fence driver on ring 10 use gpu addr 0x00000000004000ac, cpu addr 0xffff8807e4c8a0ac
[    6.564157] amdgpu 0000:29:00.0: fence driver on ring 11 use gpu addr 0x00000000004000bc, cpu addr 0xffff8807e4c8a0bc
[    6.565147] [drm] Found UVD firmware Version: 1.79 Family ID: 16
[    6.565467] amdgpu 0000:29:00.0: fence driver on ring 12 use gpu addr 0x000000f4002ad420, cpu addr 0xffffc90008e5a420
[    6.566011] [drm] Found VCE firmware Version: 52.4 Binary ID: 3
[    6.566135] amdgpu 0000:29:00.0: fence driver on ring 13 use gpu addr 0x00000000004000dc, cpu addr 0xffff8807e4c8a0dc
[    6.566238] amdgpu 0000:29:00.0: fence driver on ring 14 use gpu addr 0x00000000004000ec, cpu addr 0xffff8807e4c8a0ec
[    6.612194] amdgpu: [powerplay] Can't find requested voltage id in vdd_dep_on_sclk table!
[    6.622844] [drm] DAL is enabled
[    6.623012] [drm] DM_PPLIB: values for Engine clock
[    6.623069] [drm] DM_PPLIB:	 30000
[    6.623123] [drm] DM_PPLIB:	 60800
[    6.623177] [drm] DM_PPLIB:	 91900
[    6.623230] [drm] DM_PPLIB:	 108800
[    6.623284] [drm] DM_PPLIB:	 115600
[    6.623338] [drm] DM_PPLIB:	 120300
[    6.623391] [drm] DM_PPLIB:	 124800
[    6.623445] [drm] DM_PPLIB:	 127900
[    6.623499] [drm] DM_PPLIB: Warning: using default validation clocks!
[    6.623559] [drm] DM_PPLIB: Validation clocks:
[    6.623616] [drm] DM_PPLIB:    engine_max_clock: 72000
[    6.623673] [drm] DM_PPLIB:    memory_max_clock: 80000
[    6.623730] [drm] DM_PPLIB:    level           : 0
[    6.623787] [drm] DM_PPLIB: reducing engine clock level from 8 to 2
[    6.623847] [drm] DM_PPLIB: values for Memory clock
[    6.623904] [drm] DM_PPLIB:	 30000
[    6.623958] [drm] DM_PPLIB:	 200000
[    6.624011] [drm] DM_PPLIB: Warning: using default validation clocks!
[    6.624071] [drm] DM_PPLIB: Validation clocks:
[    6.624127] [drm] DM_PPLIB:    engine_max_clock: 72000
[    6.624184] [drm] DM_PPLIB:    memory_max_clock: 80000
[    6.624241] [drm] DM_PPLIB:    level           : 0
[    6.624298] [drm] DM_PPLIB: reducing memory clock level from 2 to 1
[    6.624358] [drm] DC: create_links: connectors_num: physical:5, virtual:0
[    6.624421] [drm] Connector[0] description:signal 32
[    6.624480] [drm] Using channel: CHANNEL_ID_DDC1 [1]
[    6.624552] [drm] Connector[1] description:signal 32
[    6.624610] [drm] Using channel: CHANNEL_ID_DDC3 [3]
[    6.624681] [drm] Connector[2] description:signal 32
[    6.624739] [drm] Using channel: CHANNEL_ID_DDC2 [2]
[    6.624810] [drm] Connector[3] description:signal 4
[    6.624868] [drm] Using channel: CHANNEL_ID_DDC4 [4]
[    6.624939] [drm] Connector[4] description:signal 2
[    6.624997] [drm] Using channel: CHANNEL_ID_DDC6 [6]
[    6.635261] [drm] Display Core initialized
[    6.635352] [drm] amdgpu: freesync_module init done ffff8807fa70dcc0.
[    6.635651] [drm] link=0, dc_sink_in=          (null) is now Disconnected
[    6.635713] [drm] DCHPD: connector_id=0: dc_sink didn't change.
[    6.635879] [drm] link=1, dc_sink_in=          (null) is now Disconnected
[    6.635941] [drm] DCHPD: connector_id=1: dc_sink didn't change.
[    6.648146] [drm] [Detect]	[DP][ConnIdx:2] Rx Caps: 12 14 C4 01 01 00 01 00 02 02 06 00 00 00 00 ^
[    6.659828] [drm] [LKTN]	[DP][ConnIdx:2] HBR2x4 pass VS=1, PE=1^
[    6.669581] [drm] [Detect]	[DP][ConnIdx:2] PL4071UH: [Block 0] 00 FF FF FF FF FF FF 00 26 CD 09 00 01 01 01 01 00 19 01 04 B5 58 31 78 3A 0D 4D AB 4F 42 A6 26 0E 47 4A BF EF 80 E1 C0 D1 00 D1 C0 B3 00 A9 40 A9 C0 81 80 81 00 4D D0 00 A0 F0 70 3E 80 30 20 35 00 6E E5 31 00 00 1A 04 74 80 18 71 70 5A 80 58 2C 8A 00 6E E5 31 00 00 1E 00 00 00 FD 00 17 4C 0F 8A 3C 00 0A 20 20 20 20 20 20 00 00 00 FC 00 50 4C 34 30 37 31 55 48 0A 20 20 20 20 01 3B ^
[    6.669733] [drm] [Detect]	[DP][ConnIdx:2] PL4071UH: [Block 1] 02 03 2E F3 53 10 1F 05 14 04 13 03 02 12 11 07 06 16 15 20 01 DD DE DF 23 09 7F 07 83 01 00 00 6D B9 14 00 04 00 B8 6E 20 00 60 01 02 03 04 74 00 30 F2 70 5A 80 B0 58 8A 00 6E E5 31 00 00 1E 56 5E 00 A0 A0 A0 29 50 30 20 35 00 6E E5 31 00 00 1A 02 3A 80 18 71 38 2D 40 58 2C 45 00 6E E5 31 00 00 1E 01 1D 00 72 51 D0 1E 20 6E 28 55 00 6E E5 31 00 00 1E 00 00 00 00 00 00 00 00 00 D4 ^
[    6.669878] [drm] dc_link_detect: manufacturer_id = CD26, product_id = 9, serial_number = 1010101, manufacture_week = 0, manufacture_year = 25, display_name = PL4071UH, speaker_flag = 1, audio_mode_count = 1
[    6.669969] [drm] dc_link_detect: mode number = 0, format_code = 1, channel_count = 1, sample_rate = 127, sample_size = 7
[    6.670043] [drm] link=2, dc_sink_in=ffff8807ee630400 is now Connected
[    6.670103] [drm] DCHPD: connector_id=2: Old sink=          (null) New sink=ffff8807ee630400
[    6.670694] [drm] link=3, dc_sink_in=          (null) is now Disconnected
[    6.670755] [drm] DCHPD: connector_id=3: dc_sink didn't change.
[    6.670893] [drm] link=4, dc_sink_in=          (null) is now Disconnected
[    6.670959] [drm] DCHPD: connector_id=4: dc_sink didn't change.
[    6.671027] [drm] Supports vblank timestamp caching Rev 2 (21.10.2013).
[    6.671088] [drm] Driver supports precise vblank timestamp query.
[    6.671147] [drm] KMS initialized.
[    6.672600] [drm] ring test on 0 succeeded in 11 usecs
[    6.673286] [drm] ring test on 9 succeeded in 7 usecs
[    6.673364] [drm] ring test on 1 succeeded in 7 usecs
[    6.673430] [drm] ring test on 2 succeeded in 2 usecs
[    6.673496] [drm] ring test on 3 succeeded in 2 usecs
[    6.673561] [drm] ring test on 4 succeeded in 2 usecs
[    6.673627] [drm] ring test on 5 succeeded in 2 usecs
[    6.673693] [drm] ring test on 6 succeeded in 2 usecs
[    6.673759] [drm] ring test on 7 succeeded in 2 usecs
[    6.673825] [drm] ring test on 8 succeeded in 2 usecs
[    6.673959] [drm] ring test on 10 succeeded in 5 usecs
[    6.674025] [drm] ring test on 11 succeeded in 5 usecs
[    7.690209] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[    8.700434] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[    9.710656] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[   10.720882] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[   11.731101] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[   12.741335] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[   13.751560] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[   14.761781] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[   15.772007] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[   16.782237] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[   16.802139] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, giving up!!!
[   17.093900] [drm:uvd_v6_0_ring_test_ring [amdgpu]] *ERROR* amdgpu: ring 12 test failed (0xCAFEDEAD)
[   17.093993] [drm:amdgpu_device_init [amdgpu]] *ERROR* hw_init of IP block <uvd_v6_0> failed -22
[   17.094062] amdgpu 0000:29:00.0: amdgpu_init failed
...
```

**Comment by gstoner** [comment]
We currently only testing Fedora 24,  so bit ahead of the curve with 26,  one thing the 4.11 kernel had firmware issue with Fedora,  they are looking at addressing it in 1.6.2

**Comment by oheid** [comment]
A curious observation: The first (warm) reboot into 4.11.0-kfd+ after building it under 4.9.0-kfd+ usually succeeds, but never a second reboot or cold boot.
I use the latest amdgpu and radeon firmware trees, by replacing the default ones in /lib/firmware.

**Comment by fxkamd** [comment]
This is not just a problem with X11. If the GPU initialization fails in this way it will break ROCm as well.

You can try skipping the UVD initialization with amdgpu.ip_block_mask=0xfffffeff

**Comment by oheid** [comment]
Thanks! I tried your suggestion but without success. Is there a point in trying other masks?

**Comment by fgarpe** [comment]
The same problem here with latest amdgpu-pro drivers and latest rocm kernel :(
[    7.081328] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[    8.093626] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[    9.105914] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[   10.118206] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[   11.130495] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[   12.142786] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[   13.155084] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[   14.167380] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[   15.179676] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[   16.191969] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, trying to reset the VCPU!!!
[   16.211918] [drm:uvd_v6_0_start [amdgpu]] *ERROR* UVD not responding, giving up!!!
[   16.413022] [drm:uvd_v6_0_ring_test_ring [amdgpu]] *ERROR* amdgpu: ring 12 test failed (0xCAFEDEAD)
[   16.413114] [drm:amdgpu_device_init [amdgpu]] *ERROR* hw_init of IP block <uvd_v6_0> failed -22

**Comment by kentrussell** [comment]
Can you try the 1.9 release and see if the issue persists?

**Comment by jlgreathouse** [comment]
With 2.0 released, and no response since before 1.9, I'm going to be an optimist and assume this has been fixed.
